src/api/readwise.rs

```diff
@@ -0,0 +1,172 @@
+// This is free and unencumbered software released into the public domain.
+
+use crate::api::types::{BookListResponse, HighlightsResponse, ReadwiseConfig};
+use anyhow::Result;
+use reqwest::Client;
+
+pub struct ReadwiseClient {
+    config: ReadwiseConfig,
+    client: Client,
+}
+
+impl ReadwiseClient {
+    pub fn new(config: ReadwiseConfig) -> Result<Self> {
+        let client = Client::builder()
+            .timeout(std::time::Duration::from_secs(config.timeout))
+            .build()?;
+
+        Ok(Self { config, client })
+    }
+
+    fn auth_header(&self) -> String {
+        format!("Token {}", self.config.access_token)
+    }
+
+    pub fn endpoint_url(&self, path: &str) -> String {
+        self.config.endpoint_url(path)
+    }
+
+    fn build_url_with_params(
+        &self,
+        path: &str,
+        page_size: Option<usize>,
+        page: Option<usize>,
+    ) -> String {
+        let mut url = self.endpoint_url(path);
+        let mut params = vec![];
+
+        if let Some(size) = page_size {
+            params.push(format!("page_size={}", size));
+        }
+        if let Some(p) = page {
+            params.push(format!("page={}", p));
+        }
+
+        if !params.is_empty() {
+            url.push('?');
+            url.push_str(&params.join("&"));
+        }
+
+        url
+    }
+
+    pub async fn fetch_highlights(
+        &self,
+        page_size: Option<usize>,
+        page: Option<usize>,
+    ) -> Result<HighlightsResponse> {
+        let url = self.build_url_with_params("/highlights/", page_size, page);
+        let response = self
+            .client
+            .get(&url)
+            .header("Authorization", self.auth_header())
+            .send()
+            .await?;
+
+        let response_text = response.text().await?;
+        let highlights: HighlightsResponse = serde_json::from_str(&response_text)?;
+        Ok(highlights)
+    }
+
+    pub async fn fetch_booklist(
+        &self,
+        page_size: Option<usize>,
+        page: Option<usize>,
+    ) -> Result<BookListResponse> {
+        let url = self.build_url_with_params("/books/", page_size, page);
+        let response = self
+            .client
+            .get(&url)
+            .header("Authorization", self.auth_header())
+            .send()
+            .await?;
+
+        let response_text = response.text().await?;
+        let booklist: BookListResponse = serde_json::from_str(&response_text)?;
+        Ok(booklist)
```

Comment:
There's a `.json()` helper in `reqwest` (behind the feature gate `json` that you're importing)

```suggestion
        Ok(self
            .client
            .get(&url)
            .header("Authorization", self.auth_header())
            .send()
            .await?
            .json()
            .await?)
```

Reply:
it makes sense. I saw other module and reqwest is not used, because of that, I switched from reqwest to ureq